tools/boilerplate.py

```diff
@@ -379,6 +382,78 @@ def build_pyplot(pyplot_path):
         pyplot.writelines(pyplot_orig)
         pyplot.writelines(boilerplate_gen())
 
+    # Run black to autoformat pyplot
+    subprocess.run(
+        [sys.executable, "-m", "black", "--line-length=79", pyplot_path]
+    )
```

Comment:
Certainly it is possible, but the tradeoff would be that the generation code would be less readable (and possibly the output code as well, though in a more subjective way)

If I were to write it, I would probably at least start with black-style (even if generated ourselves) for the signatures, rather than using `textwrap` (but perhaps keeping `textwrap` for the method bodies)... with type hints each variable (and the return annotation ) being on a separate line is much more readable, as it is often less obvious where a new variable starts, especially with types that include commas. (I would probably still try to compress to a single line if it'll fit, but that is relatively rare with the added length of the annotations).
While relatively rare, "one line per parameter" is not always itself sufficient to guarantee short enough lines, so that requires additional handling.
I'm not over the moon about how `black` handles these cases. I think it is a weak point of `black`'s style. But it's probably still better at handling these edge cases than most formatting code that is itself readable/short enough to not be totally lost in formatting weeds.

The fact of the matter is that I was apprehensive about writing it even with the list of edge cases I saw quickly, and I'm sure I would discover at least a few more while implementing. `black` was quite honestly the quickest answer to the problems I had at hand, so I went with it.

I don't think the bump to 88 will significantly impact whether `textwrap` will do the trick... there are far too many of these calls that span multiple lines and so while it will shift which ones get cut off on the ` -> `, I doubt it will get rid of them (and certainly even if we get lucky, it will be fragile to future change)
It _is_ more likely to ensure that more of the type hints fit on a single line, but still won't get some of the larger Unions/parameters with `Callable` type hints.
I can also think of other ways `textwrap` is fragile that are more likely to be problematic with type hints. For example, if a string with a space appears in the signature (not something we commonly do, but some of the positioning code for things like legend take strings like `"upper right"`, so if we decided to encode that into a Literal type or use that as the default, there would be nothing preventing the `textwrap` based code formatting from breaking on that space, which is not syntactically valid.
We could probably get away without addressing these edge cases for quite a long time, but eventually are likely to need something more syntax aware in more and more ways.


In any case, the move to `black` for pyplot was primarily one of not wanting to get lost in minutia of how autogenerated code is formatted while retaining correct syntax. 
I think the generation code is easier to understand, as I was able to get rid of some of the other workarounds (like using a `\0` to avoid breaking on the data arg dict unpacking, which I could also just do something similar for ` -> ` if my only goal was "<X characters and syntactically valid")

It now has in the generation code with only the "syntactically valid" code and then a clean up at the end to do both "<X characters" and "mostly readable", without breaking "syntactically valid". Those are precisely the _goals_ of `black`(we can argue about the degree to which it achieves it, but it is the goal).
While readability is _less_ useful for autogenerated code than handwritten code, I'd argue that it still matters, especially for signatures.


Wow, that got away from me a bit more than I expected when I started writing... tl;dr yes black is perhaps overkill, but type hints complicate signatures in ways that invite formatting edge cases, so I expect any solution that doesn't involve an extra dependency to be more fragile and harder to follow and update the generation code.